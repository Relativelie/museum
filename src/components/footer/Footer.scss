@import "../../variables";

.footer {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-column-gap: $distanceBetweenChild;
    margin: 50px $defaultMargin 80px $defaultMargin;
}

// TEXT
.footer_content {
    display: flex;
    flex-direction: column;
}

.footer_content_title {
    @include menuTextSettings($fontSize: 16px, $lineHeight: 23.76px);
    margin: 0 0 10px 0;
}

.footer_content__text, .footer_content__anchor {
    @include menuTextSettings($fontSize: 14px, $lineHeight: 20.79px);
    margin: 0 0 25px 0;
}

// SOCIAL IMAGES
.footer_social {
    display: flex;
    align-items: center;

    & img {
        width: 27.9px;
        height: 27.9px;
        margin-right: $distanceBetweenChild;
        border-radius: 10px;
        transform: box-shadow 1s ease;
        cursor: pointer;

        &:hover {
            box-shadow: 0 1px 10px #FFFFFF;
        }
    }
}

@media all and (max-width: 800px) {
    .footer {
        display: flex;
        flex-direction: column;
        align-items: center;
    }
}
