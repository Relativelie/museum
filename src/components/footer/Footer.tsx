import vk from '../../assets/images/footer/VK.comiconsvg.svg';
import tiktok from '../../assets/images/footer/tiktokicon.png';
import youtube from '../../assets/images/footer/youtube.png';
import facebook from '../../assets/images/footer/Facebookicon.png';

import './Footer.scss';

export const Footer = () => {
    const link = 'https://www.figma.com/file/AJg92P4NRV0azZ31k5KLIY/%D0%A2%D0%B5%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B5-%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5-%D0%B4%D0%BB%D1%8F-%D1%80%D0%B0%D0%B7%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%87%D0%B8%D0%BA%D0%B0?node-id=12%3A42';
    return (
        <footer className="footer">
            <div className="footer_content">
                <header>
                    <h3 className="footer_content_title">Заголовок</h3>
                </header>
                <p className="footer_content__text">ОписаниОписание Описание  Описание Описание Описание  Описаниее</p>
                <address>
                    <a className="footer_content__anchor" href={link}>500na700.ru</a>
                </address>
            </div>
            <div className="footer_social">
                <img src={vk} alt="vk" />
                <img src={facebook} alt="facebook" />
                <img src={tiktok} alt="tiktok" />
                <img src={youtube} alt="youtube" />
            </div>
        </footer>
    );
};
