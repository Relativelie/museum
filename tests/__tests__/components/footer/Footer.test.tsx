import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import { act, create } from 'react-test-renderer';

import { Footer } from '../../../../src/components/footer/Footer';

describe('Feedback - Time component', () => {
    test('snapshot - time component', () => {
        let root: any;
        act(() => {
            root = create(<Footer />);
        });
        expect(root.toJSON()).toMatchSnapshot();
    });
});
