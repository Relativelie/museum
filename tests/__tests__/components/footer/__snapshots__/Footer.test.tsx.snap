// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback - Time component snapshot - time component 1`] = `
<div
  className="footer"
>
  <div
    className="footer_content"
  >
    <h3
      className="footer_content_title"
    >
      Заголовок
    </h3>
    <p
      className="footer_content__text"
    >
      ОписаниОписание Описание  Описание Описание Описание  Описаниее
    </p>
    <a
      className="footer_content__anchor"
      href="https://www.figma.com/file/AJg92P4NRV0azZ31k5KLIY/%D0%A2%D0%B5%D1%81%D1%82%D0%BE%D0%B2%D0%BE%D0%B5-%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5-%D0%B4%D0%BB%D1%8F-%D1%80%D0%B0%D0%B7%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%87%D0%B8%D0%BA%D0%B0?node-id=12%3A42"
    >
      500na700.ru
    </a>
  </div>
  <div
    className="footer_social"
  >
    <img
      alt="vk"
      src="test-file-stub"
    />
    <img
      alt="facebook"
      src="test-file-stub"
    />
    <img
      alt="tiktok"
      src="test-file-stub"
    />
    <img
      alt="youtube"
      src="test-file-stub"
    />
  </div>
</div>
`;
